devtools/etdump/etdump_flatcc.cpp

```diff
@@ -497,27 +499,15 @@ ETDumpResult ETDumpGen::get_etdump_data() {
 }
 
 void ETDumpGen::set_debug_buffer(Span<uint8_t> buffer) {
-  debug_buffer_ = buffer;
+  data_sink_ = std::make_shared<BufferDataSink>(buffer);
```

Comment:
Curious why `make_shared` here. This is only owned internally by this class right?

Reply:
just the underlying data_sink_ is a shared_ptr;

Reply:
Our user may still want to use their`data_sink` after `ETDumpGen` lifecycle, so that I make `data_sink_` as `shared_ptr`.